### PR TITLE
Feature/automatic logout

### LINKED ISSUE #65: Close session if 401 forbidden is returned from API.
While using tokens from device_token_auth for session management, implement feature to invalidate the current session if error code 401 is returned in any request.
That way, when the token expires the user is not longer stuck with an invalid session.

### LINKED ISSUE #70: Add automatic logout when devise token expires
Once the devise token expires in the backend, the ios app should be automatically logged out.

Right now when the token expires, the app gets stuck as the session doesn't close and the user is not able to do any action (including logout)

### PR DESCRIPTION
**Pull request by glm4**
1. Refactored `SessionDataManager` and `UserDataManager`.
  -Renamed `SessionDataManager` to `SessionManager`.
  -Changed class methods to static vars with custom getters and setters.
  -Improved current session validation.
  -Updated test for new session validation.

2. Added automatic logout feature.
  -Clearing local user data and session when 401 forbidden is received from API.

Fixes #65, #70

### CODE REVIEW

File: ios-base/Manager/SessionManager.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
+//
+//  SessionDataManager.swift
+//  ios-base
+//
+//  Created by Juan Pablo Mazza on 11/8/16.
```
Comment (MaicolBen) [inline ios-base/Manager/SessionManager.swift]: I think this should be removed, what do you think?

Comment (glm4) [reply ios-base/Manager/SessionManager.swift]: Would you change all files creators to Rootstrap? or just remove the ones that are no longer part of the organization?

Comment (MaicolBen) [reply ios-base/Manager/SessionManager.swift]: After discussing with @matiasmansilla1989, we don't see anything valuable to have this, you can create another PR removing this line in all the files

Comment (glm4) [reply ios-base/Manager/SessionManager.swift]: @MaicolBen these lines are generated automatically by Xcode when creating new files.

Comment (MaicolBen) [reply ios-base/Manager/SessionManager.swift]: Let it as it is, thanks

**Pull request merged**